docs/logs.md

```diff
@@ -0,0 +1,67 @@
+2) Use Descriptive Severity Levels: Use descriptive severity levels to indicate the importance of a log message. For example, use "INFO" for informational messages, "WARN" for problems that do not interfere with the final result of the request, and "ERROR" for errors that affect the final result of the request.
+3) Include Relevant Information: Logs should include relevant information about the action being done.
+4) Use Structured Logging: Structured logging is a technique that organizes log messages into a structured format. Structured logging enables users to search, filter, and analyze logs more efficiently.
+5) Provide Contextual Information: When possible, provide contextual information that helps users to understand the log message. For example, if the log message indicates that a request has failed, provide additional information about why the request has failed, such as a network error, unsupported dependency, or a malformed input parameter.
+6) Use Clear and Concise Language: Use clear and concise language to describe the log messages. Avoid using technical jargon or abbreviations that users may not understand.
+
+
+## Logging conventions
+
+In our logs, we will use structured log messages, partially formatted in JSON with key-value pairs as described below.
+
+```
+TIMESTAMP LOG_LEVEL LOGGER MESSAGE ADDITIONAL_JSON_DATA
+```
+
+For example:
+
+```
+2023-03-07T11:32:29.948Z WARN pkg_managers.pip No setup.py found in directory /home/fepas/cachi2/workdir/cachito-pip-with-deps, package is likely not pip compatible
+2023-03-07T11:32:29.948Z INFO pkg_managers.pip Filling in missing metadata from setup.cfg
+2023-03-07T11:32:29.949Z INFO pkg_managers.pip Found metadata {"name": "cachito-pip-with-deps", "version": "1.0.0" }
+2023-03-07T11:32:29.949Z INFO pkg_managers.pip Resolved package {"name": "cachito-pip-with-deps", "version": "1.0.0"}
+2023-03-07T11:32:29.951Z INFO pkg_managers.pip No hash options used, will not require hashes unless HTTP(S) dependencies are present.
+2023-03-07T11:32:29.959Z INFO pkg_managers.pip Dependencies found [{"name": "ruamel-yaml-clib", "version": "0.2.6"}, {"name": "six", "version": "1.16.0"}, {"name": "urllib3", "version": "1.26.7"}, {"name": "cython", "version": "0.29.33"}]
+2023-03-07T11:32:29.161Z INFO pkg_managers.pip Downloading dependencies [{"name": "ruamel-yaml-clib", "version": "0.2.6"}, {"name": "six", "version": "1.16.0"}, {"name": "urllib3", "version": "1.26.7"}, {"name": "cython", "version": "0.29.33"}]
+2023-03-07T11:32:29.161Z INFO pkg_managers.pip Dependencies downloaded successfully [{"name": "ruamel-yaml-clib", "version": "0.2.6"}, {"name": "six", "version": "1.16.0"}, {"name": "urllib3", "version": "1.26.7"}, {"name": "cython", "version": "0.29.33"}]
```

Comment:
Should we log the entire list of dependencies on a single line? It already gets long with 3 deps, imagine npm requests with 2000 dependencies

And we probably only need to log them once, not 3 times as "found -> downloading -> downloaded"